### PR TITLE
Locally uniform comparison image descriptor

### PR DESCRIPTION
**Pull request by Str3iber**
Provides a free implementation of LUCID (Ziegler, Andrew, Eric Christiansen, David Kriegman, and Serge J. Belongie. "Locally uniform comparison image descriptor." In Advances in Neural Information Processing Systems, pp. 1-9. 2012.). LUCID should be much faster, while about as robust, or slightly less than, e.g. SURF or BRIEF. This is compatible with BFMatcher and the code was optimized for OpenCV with the help of gliese581h. This commit includes an exposed separable averaging blur function which is required for LUCID.

### CODE REVIEW

File: modules/xfeatures2d/include/opencv2/xfeatures2d.hpp [hunk 1]
```diff
@@ -128,7 +128,26 @@ class CV_EXPORTS BriefDescriptorExtractor : public DescriptorExtractor
...
+    static Ptr<LUCID> create(const int lucid_kernel, const int blur_kernel);
+};
+
+/** @brief Separable box filter blur, needed by LUCID, also exposed for the user
+
+@param _src Image on which blur should be applied
+@param _dst Image resulting from _src having blur applied, the output image
+@param kernel Blur kernel size where 1 equates a 3x3 matrix, 2 = 5x5, 3 = 7x7, and so on
+*/
+CV_EXPORTS void separable_blur(const InputArray _src, CV_OUT OutputArray _dst, const int kernel);
```
Comment (vpisarev) [inline modules/xfeatures2d/include/opencv2/xfeatures2d.hpp]: looks like cv::blur() does the same thing. If yes, could you remove this function and use the standard one?

**Comment by vpisarev** [comment]
thank you for the contribution! Please, check my comment regarding the blur. Also, the RST documentation is deprecated now, would it be possible to put the docs as Doxygen comments?

**Comment by Str3iber** [comment]
You're welcome! Sure, I'll make the changes, thanks.

**Comment by Str3iber** [comment]
@vpisarev docs failed to merge, though I'm not quite sure why. Error looks random. Please let me know if this is what you had in mind regarding the changes.

**Comment by vpisarev** [comment]
:+1:

**Pull request merged**